src/rgw/rgw_sal_motr.cc

```diff
@@ -1932,9 +1937,17 @@ int MotrObject::write_mobj(const DoutPrefixProvider *dpp, bufferlist&& in_buffer
     if (left < bs)
       bs = this->get_optimal_bs(left);
     if (left < bs) {
-      ldpp_dout(dpp, 20) <<__func__<< " Padding [" << (bs - left) << "] bytes" << dendl;
-      data.append_zero(bs - left);
-      left = bs;
+      // This is the last I/O.
+      // Pad to allign with unit size (and not the group size) and 
+      // set M0_ENF_NO_RMW flag to force no RMW
+      mobj->ob_entity.en_flags |= M0_ENF_NO_RMW;
+      uint64_t lid = M0_OBJ_LAYOUT_ID(meta.layout_id);
+      unsigned unit_sz = m0_obj_layout_id_to_unit_size(lid);
+
+      unsigned left_aligned_to_unitsz = roundup(left, unit_sz);
+      ldpp_dout(dpp, 20) <<__func__<< " Padding [" << (left_aligned_to_unitsz - left) << "] bytes" << dendl;
+      data.append_zero(left_aligned_to_unitsz - left);
+      bs = left_aligned_to_unitsz;
```

Comment:
Looks like there is a bug here. `bs` is bigger than `left` now, so `left -= bs` in the for loop above would result in negative left. But, `left` is unsigned, so, in fact, it will be bigger than 0 and the loop will not end...

Reply:
yes, true. Need to change data type to int64_t

Reply:
Changed data type of 'bs' and 'left' to int64_t